app/Controller/CoGroupsController.php

```diff
@@ -439,7 +439,14 @@ function isAuthorized() {
     
     // Add a new Group?
     $p['add'] = ($roles['cmadmin'] || $roles['coadmin'] || $roles['comember']);
-    
+    // XXX CO-1637
```

Comment:
XXX

Also, it would be clearer to do something like

```
$p['add'] = $roles['cmadmin'] || $roles['coadmin'] || $roles[`couadmin'] 
          || ($this->...->CoSetting->nonAdminsGroupCreate(...) && $roles['comember']);
```